File: frontend/src/service/task.ts
import { GetProp, UploadProps } from 'antd';
import axios from 'axios';
import useSWR from 'swr';
import { AppInfo, ListAppTasksResponse, SubmitAppResponse } from '../model/app';
import { fetcher } from './common';

type FileType = Parameters<GetProp<UploadProps, 'beforeUpload'>>[0];


export async function submitApp(info: AppInfo) {
    const formData = new FormData();
    formData.append('jdk_version', info.jdkVersion.toString());
    formData.append('authentication_type', info.authenticationType.toString());
    formData.append('file', info.file[0].originFileObj as FileType)
    if (info.username) {
        formData.append('username', info.username);
    }

    const resp = await axios.post<SubmitAppResponse>('/api/run', formData, {
        headers: {
            'Content-Type': 'multipart/form-data'
        }
    });
    return resp.data;
}

export function useTasks(pageNo: number, pageSize: number, userID?: number) {
    let url = `/api/tasks?page_no=${pageNo}&page_size=${pageSize}`
    if (userID) {
        url += `&user_id=${userID}`;
    }
    return useSWR<ListAppTasksResponse>(url, fetcher);
}

export async function readLog(uuid: string, logType: string) {
    const resp = await axios.get<string>("/api/logs", {
        params: {
            uuid,
            log_type: logType
        }
    });
    return resp.data;
}

File: backend/pkg/model/request/task.go
package request

import (
	"mime/multipart"
)

type SubmitAppRequest struct {
	File               *multipart.FileHeader `form:"file" binding:"required"`
	JdkVersion         int32                 `form:"jdk_version" binding:"required"`
	AuthenticationType int32                 `form:"authentication_type" binding:"required"`
	Username           *string               `form:"username"`
}

type ListAppRunTasksRequest struct {
	PageNo     int    `form:"page_no" binding:"required"`
	PageSize   int    `form:"page_size" binding:"required"`
	UserID     *uint  `form:"user_id"`
	OperatorID *uint  `form:"operator_id"`
	StartTime  *int64 `form:"start_time"`
	EndTime    *int64 `form:"end_time"`
}

type GetLogRequest struct {
	LogType string `form:"log_type" binding:"required"`
	UUID    string `form:"uuid" binding:"required"`
}
